File: 1-memento/java/implementation-v1-simplified/Originator.java
public class Originator {
    private String property1;
    private String property2;

    public String getProperty1(){
        return this.property1;
    }

    public void setProperty1(String property1){
        this.property1 = property1;
    }

    public String getProperty2(){
        return this.property2;
    }

    public void setProperty2(String property2){
        this.property2 = property2;
    }

    public Memento createMemento(){
        return new Memento(this.property1, this.property2);
    }

    public void setMemento(Memento state){
        this.property1 = state.getProperty1();
        this.property2 = state.getProperty2();
    }
}

File: 1-memento/js/implementation-v1-simplified/Originator.js
const  Memento = require("./Memento.js")

module.exports = class Originator {
    #state = null;
    setState(state) {
        this.#state = state;
    }
    getState() {
        return this.#state;
    }
    createMemento() {
        return new Memento(this.#state);
    }
    restoreMemento(memento) {
        if(!(memento instanceof Memento)) throw new Error("memento must be a Memento");
        this.#state = memento.getState();
    }
}